packages/components/stamp/src/stamp.tsx

```diff
@@ -133,16 +150,42 @@ const getStampStyles = (props: StylesFunctionParams) => {
 const Stamp = (props: Props) => {
   const { themedValue } = useTheme();
   const overrideTextColor = themedValue(false, true);
+  const Icon =
+    props.icon &&
+    cloneElement(props.icon, {
+      size: 'medium',
+      color: props.tone ? iconColorsMap[props.tone] : null,
+    });
 
-  return (
+  return props.children ? (
+    <>
+      <div
+        css={[
+          getStampStyles({ ...props, overrideTextColor }),
+          getToneStyles({ ...props, overrideTextColor }),
+          getPaddingStyle(props),
+        ]}
+      >
+        {props.children}
+      </div>
+      {console.warn(
+        'Please pass messages or icons as inline props, this method will be deprecated soon. for more information, see documentation: https://uikit.commercetools.com/?path=/story/components-stamps--stamp'
+      )}
+    </>
+  ) : (
     <div
       css={[
         getStampStyles({ ...props, overrideTextColor }),
         getToneStyles({ ...props, overrideTextColor }),
         getPaddingStyle(props),
       ]}
     >
-      {props.children}
+      <SpacingsInline alignItems="center">
+        {Icon}
+        <Text.Detail tone={themedValue(undefined, 'inherit')}>
+          <FormattedMessage {...props.message} />
+        </Text.Detail>
```

Comment:
You can use the `intlMessage` prop from the `Text.Detail` component to pass it the message so you don't have to use the `FormattedMessage` component in this one.